mdbenchmark/plot.py

```diff
@@ -0,0 +1,188 @@
+    '--output-name',
+    '-o',
+    default=None,
+    help="name of output files",
+    show_default=True)
+@click.option(
+    '--output-type',
+    '-t',
+    help="file extension for plot outputs",
+    type=click.Choice(['png', 'pdf', 'svg', 'ps']),
+    show_default=True,
+    default='png')
+@click.option(
+    '--host-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="hostname",
+    show_default=True)
+@click.option(
+    '--module-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="module name or engine name (gromacs, namd)",
+    show_default=True)
+@click.option(
+    '--host-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="hostname",
+    show_default=True)
+@click.option(
+    '--gpu/--no-gpu',
+    help="plot data for GPU runs",
+    show_default=True,
+    default=False)
+@click.option(
+    '--cpu/--no-cpu',
+    help="plot data for CPU runs",
+    show_default=True,
+    default=True)
+@click.option(
+    '--plot-cores',
+    help="Plot performance per core instead of nodes",
+    show_default=True,
+    is_flag=True)
+def plot(csv, output_name, output_type, host_name, module_name, gpu, cpu, plot_cores):
+    """Generate plots from csv files"""
+
+    if len(csv) == 0:
+        console.error("No csv file specified. use --csv to specify a file.")
+
+    df = pd.concat([pd.read_csv(c, index_col=0) for c in csv]).dropna()
+
+    # gpu/cpu can be plotted together or separately
+    if gpu and cpu:
+        # this is incase the user decides to set both flags to true
+        console.info("plotting GPU and CPU data.")
+    elif gpu:
+        df = df[df.gpu]
+        console.info("plotting GPU data only.")
+    elif cpu:
+        df = df[~df.gpu]
+        console.info("plotting GPU data only.")
+    else:
+        console.info("plotting GPU and CPU data.")
+
+    for host in host_name:
+        if host in df['host'].tolist():
+            df = df[df['host'].isin(host_name)]
+            console.info('Data for the following hosts will be plotted: {}.', set(host_name))
+        elif not host:
+            # dataframe stays the same and all hosts will be plotted
+            console.info('All hosts will be plotted.')
+        elif host not in df['host'].tolist():
+            # filter host names that don't exist
+            console.error('The host {} does not exist in your csv data. Exiting.', host)
+
+    if not host_name:
+        console.info("plotting all hosts in csv.")
+
+    for module in module_name:
+        if module in ['gromacs', 'namd']:
+            console.info("plotting all modules for engine {}.", module)
+        elif module not in df['module'].tolist():
+            console.error("The module {} does not exist in your data. Exiting.", module)
+
+    if not module_name:
+        console.info("plotting all modules in csv.")
+    # this should work but we need to check before whether any of the entered
+    # names are faulty/don't exist
+    if module_name:
```

Comment:
You could piggy back the `if`-clause above an just convert that this one to `else:`